Src/AutoMoq/MockType.cs

```diff
@@ -68,11 +68,14 @@ internal static Type GetMockedType(this Type type)
             where TMock : class
         {
             return setup.Returns(() =>
-                {
-                    var result = (TResult) context.Resolve(typeof (TResult));
-                    setup.Returns(result);
-                    return result;
-                });
+            {
+                var specimen = context.Resolve(typeof (TResult));
+                TResult result = specimen is TResult ? (TResult) specimen : default(TResult);

```

Comment:
Sorry for the late reply!



> Converting OmitSpecimen to default(T) shouldn't be the responsibility of an Auto-Mocking Container.

>OmitSpecimen exists for the purpose of enabling recursion handlers and auto-property commands to skip the value. Prematurely converting the value takes away that ability.

>Thus, I'm inclined to think that this conversion is undesirable.

I agree completely.

> What is the high-level scenario where this conversion is necessary?

* The user asks the fixture to create an auto-configured mock for interface `A`.
* This interface has a method `M` that returns an instance of type `B`.
* The return value of type `B` needs to be resolved lazily, IFF the method is called.
* Ideally, we'd want AutoMoq to, somehow, tell Moq "Hey, when this method is called ask me whether I can create an instance of `B`. If i can create it, return it, otherwise fallback to whatever your default action is."

Turns out we can't ask Moq to do this. Not in a viable, fruitful way.

The best we can do, IMO, is tell Moq "Hey, when this method is called ask me whether I can create an instance of `B`. If i can create it, return it, otherwise return this other value instead (e.g., null)."